### DISCUSSION TITLE
iOS equivalent of custom activity

### DISCUSSION THREAD

**Discussion by ahosamanedinesh** [NativeScript/NativeScript#10258] [Q&A] [answered]
Hi guys,

I was looking at the documentation for creating custom android.app.Application and android.app.Activity implementations.
https://docs.nativescript.org/advanced-concepts.html#custom-application-and-activity

This feature can be very useful for passing bundle arguments. Is there an equivalent for iOS?
I am looking for ways to pass data from Native iOS to NativeScript.

**Comment by rigor789** [comment, marked as answer]
Late, but the iOS equivalent is a custom application delegate. 

```ts
import { Application } from "@nativescript/core";

@NativeClass
class MyAppDelegate extends UIResponder implements UIApplicationDelegate {
  public static ObjCProtocols = [UIApplicationDelegate];
  
  applicationDidFinishLaunchingWithOptions(
    application: UIApplication,
    launchOptions: NSDictionary<any, any>
  ): boolean {
    console.log( "applicationWillFinishLaunchingWithOptions:", launchOptions);

    return true;
  }
  
  applicationDidBecomeActive(application: UIApplication): void {
    console.log("applicationDidBecomeActive:", application);
  }
}

Application.ios.delegate = MyDelegate;
```

Something like that...
